packages/netlify-cms-core/src/actions/mediaLibrary.ts

```diff
@@ -0,0 +1,505 @@
+    }
+    dispatch({ type: MEDIA_LIBRARY_OPEN, payload });
+  };
+}
+
+export function closeMediaLibrary() {
+  return (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
+    const state = getState();
+    const mediaLibrary = state.mediaLibrary.get('externalLibrary');
+    if (mediaLibrary) {
+      mediaLibrary.hide();
+    }
+    dispatch({ type: MEDIA_LIBRARY_CLOSE });
+  };
+}
+
+export function insertMedia(mediaPath: string | string[]) {
+  return (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
+    const state = getState();
+    const config = state.config;
+    const collectionName = state.entryDraft.getIn(['entry', 'collection']);
+    const collection = state.collections.get(collectionName);
+    if (Array.isArray(mediaPath)) {
+      mediaPath = mediaPath.map(path => selectMediaFilePublicPath(config, collection, path));
+    } else {
+      mediaPath = selectMediaFilePublicPath(config, collection, mediaPath as string);
+    }
+    dispatch({ type: MEDIA_INSERT, payload: { mediaPath } });
+  };
+}
+
+export function removeInsertedMedia(controlID: string) {
+  return { type: MEDIA_REMOVE_INSERTED, payload: { controlID } };
+}
+
+export function loadMedia(
+  opts: { delay?: number; query?: string; page?: number; privateUpload?: boolean } = {},
+) {
+  const { delay = 0, query = '', page = 1, privateUpload } = opts;
+  return async (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
+    const state = getState();
+    const backend = currentBackend(state.config);
+    const integration = selectIntegration(state, null, 'assetStore');
+    if (integration) {
+      const provider = getIntegrationProvider(state.integrations, backend.getToken, integration);
+      dispatch(mediaLoading(page));
+      try {
+        const files = await provider.retrieve(query, page, privateUpload);
+        const mediaLoadedOpts = {
+          page,
+          canPaginate: true,
+          dynamicSearch: true,
+          dynamicSearchQuery: query,
+          privateUpload,
+        };
+        return dispatch(mediaLoaded(files, mediaLoadedOpts));
+      } catch (error) {
+        return dispatch(mediaLoadFailed({ privateUpload }));
+      }
+    }
+    dispatch(mediaLoading(page));
+    return new Promise(resolve => {
+      setTimeout(
+        () =>
+          resolve(
+            backend
+              .getMedia()
+              .then((files: MediaFile[]) => dispatch(mediaLoaded(files)))
+              .catch((error: { status?: number }) => {
+                console.error(error);
+                if (error.status === 404) {
+                  console.log('This 404 was expected and handled appropriately.');
+                  dispatch(mediaLoaded([]));
+                } else {
+                  dispatch(mediaLoadFailed());
+                }
+              }),
+          ),
+        delay,
```

Comment:
The `delay` was passed an argument for the `Promise` ctor instead of the `setTimeout` function.